src/components/rule-builder/conditional-logic.tsx

```diff
@@ -0,0 +1,52 @@
+import React from 'react';
+import type { Question, Schema } from '@types';
+import RuleBuilder, { type FormRule } from './rule-builder.component';
+import { v4 as uuidv4 } from 'uuid';
+
+interface ConditionalLogicProps {
+  pageIndex: number;
+  sectionIndex: number;
+  questionIndex: number;
+  question: Question;
+  schema: Schema;
+  rules: Array<FormRule>;
+  handleAddLogic: (fieldId: string) => void;
+  onSchemaChange: (schema: Schema) => void;
+}
+
+const ConditionalLogic: React.FC<ConditionalLogicProps> = ({
+  question,
+  pageIndex,
+  sectionIndex,
+  questionIndex,
+  rules,
+  schema,
+  onSchemaChange,
+  handleAddLogic,
+}) => {
+  const rulesForQuestionIndex = rules?.findIndex((rule: FormRule) => rule?.question === question?.id);
+  const rulesForQuestion =
+    rulesForQuestionIndex !== -1 && rulesForQuestionIndex !== undefined
+      ? rules?.filter((rule: FormRule) => rule?.question === question?.id)
+      : [{ id: uuidv4(), isNewRule: false, question: question?.id }];
```

Comment:
This seems like it could be re-written to be much simpler:

```suggestion
const rulesForQuestion = rules?.filter((rule: FormRule) => rule?.question === question?.id) ??
      [{ id: uuidv4(), isNewRule: false, question: question?.id }];
```